### DISCUSSION TITLE
Controlling Compressor Power in PV-mode

### DISCUSSION THREAD

**Discussion by kirschinger** [emsesp/EMS-ESP32#2062] [Ideas]
Hello everybody,
i have installed a WLW186i. When entering the service menu on the heat pump i can set a parameter for the maximum compressor power during PV-Operation (when Input 4 active). 
Can anyone let me know if this value is available in today's entities? 
I was searching all - but was not successfull yet.
I am using actually 3.7.0-dev.38 .

![20240928_050028629_iOS](https://github.com/user-attachments/assets/60a6c09d-5d71-4eb5-93d0-35ef498f9ca6)

**Comment by proddy** [comment]
You could look at https://github.com/emsesp/EMS-ESP32/issues/1717#issuecomment-2321235221 and play around with custom entities? 

The usual way to find the settings is to change it on the heat pump while monitoring in EMS-ESP

**Comment by popopirat** [comment]
I am looking for this, too! And @Vippis2000 probably as well. 
https://github.com/emsesp/EMS-ESP32/discussions/1678#discussioncomment-10852803

**Comment by kirschinger** [comment]
i have tried to log the EMS-Traffic during changing the setting in on the heat pump.
I have attached the log's - i hope somebody here is able to help for interpretation.
one log is changing the setting for max. compressor power from 6kW to 3kW
second log is changing the setting back from 3kW to 6kW.
Not sure if the telegrams are directly sent to the bus when changes applied - or if they are coming cyclic and i may have missed it.
I was going through - but i can not find any significant changes in the protocols and the content which are maybe related with this setting.
Maybe somebody could have a look or also give advice how to monitor it better to fetch the data we are searching for.

[log - 6kW to 3kW.txt](https://github.com/user-attachments/files/17268438/log.-.6kW.to.3kW.txt)
[log - 3kW to 6kW.txt](https://github.com/user-attachments/files/17268439/log.-.3kW.to.6kW.txt)

**Reply by MichaelDvP** [reply]
Looks like the value is here with divider 10:
```
001+00:51:03.857 TRACE 385415: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: 1E (offset 54)
001+00:51:04.008 TRACE 385418: [emsesp] boiler(0x08) -B-> all(0x00), HPSilentMode(0x0484), data: 1E (offset 54)
```
and
```
001+00:54:15.539 TRACE 386234: [emsesp] thermostat(0x10) -W-> boiler(0x08), HPSilentMode(0x0484), data: 3C (offset 54)
001+00:54:15.673 TRACE 386239: [emsesp] boiler(0x08) -B-> all(0x00), HPSilentMode(0x0484), data: 3C (offset 54)
```

Can you check if it works with custom entity: device 0x08, UINT, ID 484, offset 54, factor 0.1

**Reply by kirschinger** [reply]
Sakra - yes - there are the settings in HEX - and i have not seen them...

I have used these settings - it seems on the first view it is working. For Factor i have used 100 to get W - as 0,1 was not able to use due to whatever reason.

I can write & change the entity and it is afterwards also visible then on the heat pump.

![image](https://github.com/user-attachments/assets/f221262c-7145-4bb3-a4d0-a1aa19dd127e)

I will try it in the next days and report!

**Reply by Vippis2000** [reply]
I have tried these settings also on my Bosch Compress CS7000iAW and shows me a value of 0W now.

Do you also have a working setting for showing current electric power consumption?

**Reply by Vippis2000** [reply]
But I'm not able to write a value into this entity, always remains 0W

**Reply by kirschinger** [reply]
i can see the actual power from the HP in the existing Entity:
![image](https://github.com/user-attachments/assets/56a1c476-5317-4c34-a13f-9a64b5d8b0a3)
I also have an external SmartMeter on the HP and it shows the same value.

btw - the protocol seems not to be sent cyclic - but only on demand. I saw the values first when i was actively changing the setting on the HP - before i have not seen any values.
But i am using it on a WLW186i-12 E - not sure if it is same for others...

**Reply by popopirat** [reply]
> But I'm not able to write a value into this entity, always remains 0W

Same here, no response after setting the value

**Reply by MichaelDvP** [reply]
@kirschinger 
> btw - the protocol seems not to be sent cyclic - but only on demand.

The telegram is not sent cyclic, but we fetch it once a minute. If this is not working, please send a log of > 1 minute.

**Comment by kirschinger** [comment]
Report:

The settings works pretty fine on my WLW186i-12 E
Within the EMS-ESP frontend i can change the value - the heat pump is following this parameter. 
I just have seen that in the new Firmware the Entity is still included - super cool!

![image](https://github.com/user-attachments/assets/8ad0003c-3511-4eca-b29c-72d76e36ebd0)

Manually this is working fine. 
With automation i still have one problem. I am using iobroker with the EMS adapter.
I can get all entities there - the only topic i have is that the new entity is coming to iobroker as read-only.

![image](https://github.com/user-attachments/assets/66bf7c3f-9169-4325-9aff-a76e5529300a)

Does anyone have experiences what i can possibly do to get this as a writeable parameter?
